compute_sdk/tests/integration/test_serialization.py

```diff
@@ -382,3 +398,139 @@ def test_check_strategies(strategy_code, strategy_data, function, args, kwargs):
     new_result = new_fn(*new_args, **new_kwargs)
 
     assert original_result == new_result
+
+
+@pytest.mark.parametrize("disallowed_strategy", concretes.SELECTABLE_STRATEGIES)
+def test_allowed_deserializers(disallowed_strategy):
+    allowlist = [
+        strategy
+        for strategy in concretes.SELECTABLE_STRATEGIES
+        if strategy.for_code == disallowed_strategy.for_code
+        and strategy != disallowed_strategy
+    ]
+
+    assert allowlist, "expect to have at least one allowed deserializer"
+    assert disallowed_strategy not in allowlist, "sanity check"
+
+    if disallowed_strategy.for_code:
+        serializer = ComputeSerializer(allowed_code_deserializer_types=allowlist)
+        payload = disallowed_strategy().serialize(foo)
+    else:
+        serializer = ComputeSerializer(allowed_data_deserializer_types=allowlist)
+        payload = disallowed_strategy().serialize("foo")
+
+    with pytest.raises(DeserializationError) as pyt_exc:
+        serializer.deserialize(payload)
+    assert f"serializer {disallowed_strategy.__name__} is not allowed" in str(pyt_exc)
+
+
+@pytest.mark.parametrize(
+    "list, valid",
+    [
+        (["globus_compute_sdk.serialize.concretes.DillCode"], True),
+        (["my_malicious_package.my_malicious_serializer"], False),
+        (["invalid_path_1"], False),
+        (["invalid path 2"], False),
+        ([""], False),
+        (
+            [
+                "globus_compute_sdk.serialize.concretes.DillCode",
+                "globus_compute_sdk.serialize.concretes.DillCodeTextInspect",
+                "globus_compute_sdk.serialize.concretes.DillCodeSource",
+                "globus_compute_sdk.serialize.concretes.CombinedCode",
+            ],
+            True,
+        ),
+        (
+            [
+                "globus_compute_sdk.serialize.concretes.DillCode",
+                "my_malicious_package.my_malicious_serializer",
+            ],
+            False,
+        ),
+        (
+            [
+                "globus_compute_sdk.serialize.concretes.DillCode",
+                "invalid path",
+            ],
+            False,
+        ),
+    ],
+)
+def test_allowed_deserializers_checks_imports(list, valid):
+    if valid:
+        ComputeSerializer(allowed_code_deserializer_types=list)
+    else:
+        with pytest.raises(SerdeError) as pyt_exc:
+            ComputeSerializer(allowed_code_deserializer_types=list)
+        assert "is not a valid path to a strategy" in str(pyt_exc)
```

Comment:
Similar suggestion as for the previous test.  The if-branch within the test makes it a smidgen more complicated to follow; consider making two tests, one for the valids and one for the invalids.

Reply:
good point, split